### DISCUSSION TITLE
Questions on Dataset Preparation format and Model Selection for Fine-Tuning PaddleOCR for Text Recognization

### DISCUSSION THREAD

**Discussion by Shine-5705** [PaddlePaddle/PaddleOCR#13541] [Q&A]
Hi,
I’m working on fine-tuning PaddleOCR for text recognition and I have a few questions regarding the dataset preparation and model selection.

Dataset Preparation:
Could you please clarify the correct format for the dataset? Specifically, I have an image and I’m unsure if this is the correct format to use. Here’s my example:

Image: 
![table_0](https://github.com/user-attachments/assets/60b21be9-50f0-4b20-a139-88c91b5502b9)

Current dataset format: 
[train.txt](https://github.com/user-attachments/files/16405062/train.txt)

this is the label file that I will be giving for finetuning
Is this the appropriate format, or are there specific requirements or structures I need to follow? Any guidance or examples would be greatly appreciated.

Model Selection for Fine-Tuning:
I am fine-tuning to recognize the Rupee symbol. Should I use the English pretrained model or the multilingual one? If I opt for the multilingual model, will the Latin variant work for this purpose?

Custom Dictionary:[custom_dict.txt](https://github.com/user-attachments/files/16405066/custom_dict.txt)
Additionally, I plan to add a custom dictionary for better recognition. Are there any specific steps or formats I should be aware of while integrating a custom dictionary?

Thank you for your assistance!

**Comment by UserWangZz** [comment]
https://paddlepaddle.github.io/PaddleOCR/en/ppocr/model_train/recognition.html
You can reference this doc to your OCR recognition.

**Reply by Shine-5705** [reply]
In the current dataset described in the document, each image is associated with a single word for text recognition. However, I'm encountering an issue where the model is overfitting to individual words during fine-tuning. I would like to know how to create a dataset where each image contains multiple words for text recognition. Could you please provide guidance on how to achieve this?
and example would be greatly appreciated!

**Reply by GreatV** [reply]
Refer to: https://paddlepaddle.github.io/PaddleOCR/ppocr/model_train/recognition.html. Note that the English version is slightly outdated.

**Reply by GreatV** [reply]
Format the annotation files in a way that PaddleOCR can understand. For PaddleOCR, this often involves using a `.txt` file with each line containing the image path and the text label.

```
train_data/rec/train/word_001.jpg   简单可依赖
train_data/rec/train/word_002.jpg   用科技让复杂的世界更简单
```

**Reply by Shine-5705** [reply]
I checked this but for english is this correct format 
for this image 
![financial_4](https://github.com/user-attachments/assets/3e9fc5d2-1937-44ab-a24e-46bebfdba990)

this is the format for the annotation
financial_4.png	Item March 31, 2024 December 31, 2023 Dividends ₹7685 ₹5525 Goodwill ₹8693 ₹3418 Liabilities ₹9197 ₹3589 Investments ₹2107 ₹6883 Taxes ₹4096 ₹8418 Payables ₹7179 ₹9724 Payables ₹6223 ₹87 Assets ₹1899 ₹4345 Capital ₹9422 ₹3573

**Reply by GreatV** [reply]
You have to crop the images so that each one cropped image contains only one line of text. such as:

```text
cropped_image_0.png item
cropped_image_1.png March 31, 2024
cropped_image_2.png December 31, 2023
cropped_image_3.png Dividends
cropped_image_4.png ₹7685
cropped_image_5.png ₹5525
cropped_image_6.png Goodwill
cropped_image_7.png ₹8693
```

**Reply by Shine-5705** [reply]
Okay, So as I am making the dataset only so can I take the images with one word only and then use them as a dataset for finetuning to recognize rupee symbol?
also after finetuning will this be able to detect multi text images ?
like overfitting and all !

**Reply by GreatV** [reply]
I can't understand your question. It depends on your application scenario. Generally speaking, you need to prepare a dataset that is cropped and labeled by lines, rather than by words. If in your application scenario each line contains only one word, then you can organize the dataset by words.

**Reply by Shine-5705** [reply]
I am currently working on creating a dataset for fine-tuning an OCR model, specifically to enhance its ability to recognize the Rupee symbol (₹). My approach involves generating images that contain only a single word, with a focus on including the Rupee symbol in various contexts.

Before proceeding further, I have a couple of questions:

Single Word Dataset for Fine-Tuning:

If I use this dataset, which consists of images with just one word (primarily containing the Rupee symbol), for fine-tuning the OCR model, will it effectively improve the model's ability to recognize the Rupee symbol in real-world scenarios?
Generalization and Overfitting Concerns:

After fine-tuning with this specific dataset, will the model still be capable of accurately recognizing multi-word text images?
I want to ensure that the fine-tuning process doesn't cause the model to overfit to single-word images, thus potentially reducing its performance on more complex, multi-word text images.

**Reply by UserWangZz** [reply]
For your question, I think you need to prepare two data sets, one is a single-line multi-word text image, and the other is an image containing a rupee symbol with only one word. The purpose of fine-tuning is to make the data gap between inference and training as small as possible, so that good recognition results can be achieved.
